src/marqo/core/semi_structured_vespa_index/semi_structured_vespa_index.py

```
@@ -682,3 +811,153 @@ def _create_update_statement_for_updating_field(self, fields, key, val):
+                field_name = self._extract_facet_field_name(field.id, facets)
+                facets_response.setdefault(field_name, {})
+
+                for value in field.children:
+                    group_type, value_key = self._parse_value_id(value.id)
+                    processed_stats = self._process_value_stats(value.fields)
+
+                    if facets.fields[field_name].type in ["string", "array"]:
+                        if value.id == "group:string:":
+                            # Vespa's value for not found
+                            continue
+                        # values might be nested deeply if user data is stored as a.b.c
+                        current_level = facets_response[field_name]
+                        for key in value_key:
+                            if key not in current_level:
+                                current_level[key] = {}
+                            current_level = current_level[key]
+                        current_level.update(processed_stats)
+                    elif facets.fields[field_name].type in ["number"]:
+                        if any(value in [-9223372036854775808, 'NaN'] for value in processed_stats.values()):
+                            # Vespa's value for null for int and float
+                            continue
+                        if facets.fields[field_name].ranges is None:
+                            if any(value in [-9223372036854775808, 'NaN'] for value in processed_stats.values()):
+                                # Vespa's value for null
+                                continue
+                            # aggregate statistic between int and float
+                            facets_response[field_name] = self._combine_number_stats(
+                                facets_response.get(field_name, {}), processed_stats
+                            )
+                        elif group_type != "null": # the case when there is no values in range
```

What's the `else` here? If these are expected to be exhaustive, you can error out for `else` as an invariant violation (bug). This seems like it could fail silently 

i.e. what happens if `ranges is None and group_type == null and type not in ["string", "array"]`